### DISCUSSION TITLE
How to configure my CNC

### DISCUSSION THREAD

**Discussion by Avatario34** [phil-barrett/grblHAL-teensy-4.x#72] [Q&A] [answered]
Hi Phil,

I recently bought your board, have successfully assembled it with a T41, and am now building the CNC controller.

A quick question on the configuration for my CNC, it's a Lowrider V3 from V1 Engineering (https://docs.v1engineering.com/lowrider/). It's got 3 axes: X, Y and Z. The Y-axis is driven by 2 steppers, as is the Z-axis, so 5 steppers in total. X, Y1, Y2, Z1 and Z2 have got min endstops. The Lowrider auto-squares on Y and Z, so independent homing is required for Y1 and Y2 as well as Z1 and Z2.

From what I've read so far, I believe I need to amend my_machine.h like the following:

#define BOARD_T41U5XBB
...
#define Y_GANGED             1
#define Y_AUTO_SQUARE        1
#define Z_GANGED             1
#define Z_AUTO_SQUARE        1
...

Is that correct and is there anything else to configure, e.g. in /grbl/config.h?

And: I am assuming that the A-axis on your controller maps to Y2 and the B-axis to Z2, correct?

Thanks, Christian

**Comment by phil-barrett** [comment, marked as answer]
Yes, that is correct.  grblHAL assigns the ganged drives in order.  Note that you will have to reverse one of the motors in a ganged pair of motors.  Just swap a coil pair.

That is a somewhat unusual configuration (2 ganged axes) so you might want to ask more detailed questions on the [grblHAL core discussions section](https://github.com/grblHAL/core/discussions).

And, I am interested in how the Y arrangement works. It looks like it only uses one linear guide and the other side rides on the spoilboard.

**Reply by Avatario34** [reply]
Great, thanks Phil. Will reach out to the other forum soon.

**Comment by Avatario34** [comment]
I've just built the Lowrider 3, so not much to say yet. Have cut my strut plates, the YZ plates as well as the spoilboard. Got a very positive impression. A number of people have already built the LR3 and shared their experience: https://forum.v1engineering.com/c/lowrider/your-builds-lowrider/30
